website/addons/dataverse/tests/test_model.py

```diff
@@ -11,17 +11,81 @@
 
 class TestDataverseFile(DataverseAddonTestCase):
 
-    def test_dataverse_file_url(self):
+    def test_constants(self):
+        dvf = DataverseFile()
+        assert_equal('dataverse', dvf.provider)
+        assert_equal('state', dvf.version_identifier)
 
-        # Create some dataverse file
-        dvf = DataverseFile(
-            node=self.project,
-            file_id='12345',
-        )
-        dvf.save()
+    def test_path_doesnt_crash_without_addon(self):
+        dvf = DataverseFile(node=self.project, file_id='12345')
+        self.project.delete_addon('dataverse', Auth(self.user))
+
+        assert_is(self.project.get_addon('dataverse'), None)
+
+        assert_true(dvf.file_id)
+        assert_true(dvf.waterbutler_path)
+
+    def test_waterbutler_path(self):
+        dvf = DataverseFile(node=self.project, file_id='12345')
+
+        assert_equals(dvf.file_id, '12345')
+        assert_equals(dvf.waterbutler_path, '/12345')
+
+    def test_unique_identifier(self):
+        dvf = DataverseFile(node=self.project, file_id='12345')
+
+        assert_true(dvf.file_id, '12345')
+        assert_equals(dvf.unique_identifier, '12345')
+
+    def test_node_addon_get_or_create(self):
+        dvf, created = self.node_settings.find_or_create_file_guid('12345')
+
+        assert_true(created)
+        assert_equal(dvf.file_id, '12345')
+        assert_equal(dvf.waterbutler_path, '/12345')
+
+    def test_node_addon_get_or_create_finds(self):
+        dvf1, created1 = self.node_settings.find_or_create_file_guid('12345')
+        dvf2, created2 = self.node_settings.find_or_create_file_guid('12345')
+
+        assert_true(created1)
+        assert_false(created2)
+        assert_equals(dvf1, dvf2)
+
+    @mock.patch('website.addons.base.requests.get')
```

Comment:
This is fine, but in the future, you can use `httpretty` to mock HTTP responses. An example is here: https://github.com/CenterForOpenScience/osf.io/blob/70f83e84cbd70c6bb021e6b8f874f7f67e798f82/tests/test_identifiers.py#L141-153 .

This allows for finer control over HTTP mocking and also doesn't rely on the `requests` library.